packages/components/src/RichTooltip/RichLayout/RichLayout.component.js

```diff
@@ -37,21 +37,32 @@ const Footer = ({ id, children, className }) => (
 );
 Footer.propTypes = TooltipPropTypes;
 
-export default function RichLayout(props) {
-	return (
-		<span id={props.id}>
-			<Header id={`${props.id}-header`}>
-				{Inject.getReactElement(props.getComponent, props.Header)}
-			</Header>
-			<Body id={`${props.id}-body`} className={props.className}>
-				{props.text && <p>{props.text}</p>}
-				{!props.text && Inject.getReactElement(props.getComponent, props.Content)}
-			</Body>
-			<Footer id={`${props.id}-footer`}>
-				{Inject.getReactElement(props.getComponent, props.Footer)}
-			</Footer>
-		</span>
-	);
+export default class RichLayout extends React.Component {
```

Comment:
why do you need a class here?

Reply:
https://fr.reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-class-component.
https://fr.reactjs.org/docs/refs-and-the-dom.html#refs-and-function-components

My components where is set my RichLayout call a focusRichLayout when i lost the focus on the button.

Reply:
Thank you for the documentation but my comment was more about "why do you change the functional component in favor of a class-based component?"

Reply:
If i want to access to the method instance, i need a class based component instead a function component.